git-tar/function/ops_test.go

```diff
@@ -0,0 +1,102 @@
+package function
+
+import (
+	"testing"
+
+	// internal dependencies
+	"fmt"
+	"github.com/openfaas/openfaas-cloud/sdk"
+)
+
+func genPushEvent(id int, cloneURL string, private bool) sdk.PushEvent {
+	return sdk.PushEvent{
+		Installation: sdk.PushEventInstallation{
+			ID: id,
+		},
+		Repository: sdk.PushEventRepository{
+			CloneURL: cloneURL,
+			Private:  private,
+		},
+	}
+}
+
+var (
+	defaultToken          = "authToken1234"
+	defaultInstallationID = 12345
+	defaultError          = fmt.Errorf("error message")
+)
+
+type authTokenMock struct {
+	getTokenCalledTimes int
+	returnError         bool
+}
+
+func (t *authTokenMock) GetToken() (string, error) {
+	t.getTokenCalledTimes++
+
+	if t.returnError {
+		return "", defaultError
+	}
+
+	return defaultToken, nil
+}
+
+func (t *authTokenMock) GetInstallationID() int {
+	return defaultInstallationID
+}
+
+func Test_getRepositoryURL_whenRepositoryIsNotPrivate(t *testing.T) {
+	expected := "https://foo.bar/baz"
+	pe := genPushEvent(12345, expected, false)
+	at := &authTokenMock{}
+
+	got, err := getRepositoryURL(pe, at)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if got != expected {
+		t.Errorf("Expected: %s, Got: %s", expected, got)
+	}
+}
+
+func Test_getRepositoryURL_whenRepositoryIsPrivate(t *testing.T) {
+	pe := genPushEvent(12345, "https://foo.bar/baz", true)
+	at := &authTokenMock{}
```

Comment:
Since this is not a mock but a stub/fake let's change the name. test or fakeAuthToken may work for instance.